kconfig: fix (re)definitions in Kconfig.defconfig files

gmarull opened the pull request:
Kconfig symbols must not be defined in Kconfig.defconfig.* files. Such
files are meant to change the default value of certain options.

This PR resolves such issues and introduces a new compliance check to prevent similar cases in the future.

gmarull committed:
boards: shields: ls0xx_generic: remove redefined Kconfig prompts

Kconfig entries used for changing an option default value should not
re-define the option prompt. We should probably introduce a check for
that (any option has a single prompt and help).

Signed-off-by: Gerard Marull-Paretas <gerard.marull@nordicsemi.no>

tejlmand reviewed (approved):
Thanks for cleaning this up.

Maybe a simple search in all `Kconfig.defconfig` for text strings in quotes `"Some text"` should be possible, cause no defconfigs ought to have prompts.

gmarull committed:
soc: arm: nuvoton_npcx: remove Kconfig help re-definitions in defconfig

The SoC Kconfig.defconfig.series files re-defined multiple Kconfig help
entries. The help for such symbols should only be present in places
where symbol is actually defined, not where the defaults are changed.

Signed-off-by: Gerard Marull-Paretas <gerard.marull@nordicsemi.no>

gmarull committed:
soc: arm: microchip_mec: remove re-defined Kconfig help

The help for MCHP_ECIA_XEC is defined in
drivers/interrupt_controller/Kconfig.xec. defconfig files must not
re-define prompts or help, just modify the default value.

Signed-off-by: Gerard Marull-Paretas <gerard.marull@nordicsemi.no>

gmarull committed:
soc: arm64: xenvm: move Kconfig option to soc

The XEN_INITIAL_DOMAIN option was defined at board level, however, some
drivers such as the XEN serial driver has dependencies on
XEN_INITIAL_DOMAIN, meaning any new board has to define its own copy of
XEN_INITIAL_DOMAIN. Move the option to SoC Kconfig.

Signed-off-by: Gerard Marull-Paretas <gerard.marull@nordicsemi.no>

gmarull committed:
boards: riscv: rv32m1_vega: remove redundant Kconfig comment

The defconfig files are just used to change defaults, not to define any
new Kconfig nodes.

Signed-off-by: Gerard Marull-Paretas <gerard.marull@nordicsemi.no>

gmarull committed:
boards: arm: qemu_cortex_m0: move Kconfig definition to Kconfig

A board-level Kconfig option was defined in the Kconfig.defconfig file.
Move it to a board-level Kconfig file.

Signed-off-by: Gerard Marull-Paretas <gerard.marull@nordicsemi.no>

gmarull committed:
scripts: ci: check_compliance: check for (re)defined Kconfig symbols

Kconfig symbols must not be defined in Kconfig.defconfig.* files. Such
files are meant to change the default value of certain options.

Signed-off-by: Gerard Marull-Paretas <gerard.marull@nordicsemi.no>

gmarull commented:
@mbolivar-nordic @tejlmand please re-review.

tejlmand reviewed (approved):
thanks for this cleanup.

stephanosio reviewed (approved):
Nice work

jfischer-no commented on boards/arm/bl654_usb/Kconfig.defconfig:
```diff
@@ -26,24 +26,14 @@ config FLASH_LOAD_OFFSET
 config USB_CDC_ACM
 	default n if USB_DEVICE_BLUETOOTH
 
-config BL654_USB_SERIAL_BACKEND_CDCACM
```
It is not just shortcut, please move this option to boards/arm/bl654_usb/Kconfig, or drop this patch and assign an issue to me, thanks.

gmarull replied:
thanks, moved to Kconfig.

gmarull committed:
boards: arm: bl654_usb: move Kconfig definition to Kconfig

A Kconfig option was defined in the Kconfig.defconfig file. Move it to
the board Kconfig file since definitions are not allowed in defconfig
files, just change default values.

Signed-off-by: Gerard Marull-Paretas <gerard.marull@nordicsemi.no>

jfischer-no reviewed (approved):
:+1:
